bigquery_etl/routine/parse_routine.py

```diff
@@ -257,68 +240,57 @@ def parse_routines(project_dir):
     raw_routines = read_routine_dir(
         project_dir,
         Path(ConfigLoader.get("default", "sql_dir", fallback="sql")) / "mozfun",
-        ConfigLoader.get("routine", "assert_udf_dir"),
     )
 
     # prepend udf definitions to tests
     for raw_routine in raw_routines.values():
-        tests_full_sql = routine_tests_sql(raw_routine, raw_routines, project_dir)
+        tests_full_sql = routine_tests_sql(raw_routine, raw_routines)
         yield ParsedRoutine.from_raw(raw_routine, tests_full_sql)
 
 
-def accumulate_dependencies(deps, raw_routines, udf_name):
+def accumulate_dependencies(deps, raw_routines, routine_id):
     """
-    Accumulate a list of dependent routine names.
+    Accumulate a list of dependent routine IDs.
 
-    Given a dict of raw_routines and a udf_name string, recurse into the
-    routine's dependencies, adding the names to deps in depth-first order.
+    Given a dict of raw_routines and a routine_id string, recurse into the
+    routine's dependencies, adding the IDs to deps in depth-first order.
     """
-    if udf_name not in raw_routines:
+    if routine_id not in raw_routines:
         return deps
 
-    raw_routine = raw_routines[udf_name]
+    raw_routine = raw_routines[routine_id]
     for dep in raw_routine.dependencies:
         deps = accumulate_dependencies(deps, raw_routines, dep)
-    if udf_name in deps:
-        return deps
-    else:
-        return deps + [udf_name]
+        if dep not in deps:
+            deps = deps + [dep]
+    return deps
```

Comment:
This change means that the initially passed in routine will no longer be included in the list of its own dependencies (which was a bug IMO).